VictorSydorchenkoHomework1.cpp

```diff
@@ -0,0 +1,113 @@
+// Данная програмка исполняет все, указанное в задании, но я решил добавить ввод вручную, чтобы попрактиковаться.
+#include <cmath>
+#include <iostream>
+#include <stdlib.h>
+
+struct Equotion {
+  std::int64_t a{}, b{}, c{}, D{};
+  float x1{}, x2{};
+};
+
+void stringToInt(std::string &x, Equotion &res) {
+  int8_t time{0}, n{};
+  char a[5]{' ', ' ', ' ', ' ', '\0'}, b[6]{' ', ' ', ' ', ' ', ' ', '\0'},
+      c[5]{' ', ' ', ' ', ' ', '\0'};
+  for (char d : x) // main for cicle
+  {
+    if (d == 'x' || d == 'X' || d == '=' || d == 'х' || d == 'Х') {
+      time++;
+      n = 0;
+    }
+
+    if (time == 0) {
+      if (d == '-') {
+        a[n] = d;
+        n++;
+      }
+      if (d >= '0' && d <= '9') {
+        a[n] = d;
+        n++;
+      }
+    }
+    // if (time == 1)time++;
+    else if (time == 1) {
+      if (d == '-') {
+        b[n] = d;
+        n++;
+      }
+      if (d >= '0' && d <= '9') {
+        if (d == '2' && n != 0) {
+          b[n] = d;
+          n++;
+        } else if (d >= '0' && d <= '9' && d != '2') {
+          b[n] = d;
+          n++;
+        } else
+          n++;
+      }
+    } else if (time == 2) {
+      if (d == '-') {
+        c[n] = d;
+        n++;
+      }
+      if (d >= '0' && d <= '9') {
+        c[n] = d;
+        n++;
+      }
+    }
+  }
+  res.a = atoi(a);
+  res.b = atoi(b);
+  res.c = atoi(c);
+}
+
+Equotion diskrimin(Equotion &e) {
+  e.D = (e.b * e.b) - 4 * (e.a * e.c);
+  if (e.D > 0) {
+    e.x1 = (-e.b - (sqrt(e.D))) / (2 * e.a);
+    e.x2 = (-e.b + (sqrt(e.D))) / (2 * e.a);
+    return e;
+  } else if (e.D == 0) {
+    e.x1 = -e.b / 2 * e.a;
+    e.x2 = 0;
+    return e;
+  } else {
+    std::cout << "Has no roots" << std::endl;
+    return e;
+  }
+}
+bool bordersCheck(Equotion const &e) {
+  int s = 255;
+  // return 1 if borders are true
+  if (e.a > -s && e.a < s) {
+    if (e.b > -s && e.b < s) {
+      if (e.c > -s && e.c < s) {
+        return true;
+      }
+    }
+  } else {
+    return false;
+  }
+}
+
+int main() {
+  Equotion curr;
```

Comment:
можно не писать в структуре для каждого поля `{}` 
достаточно просто написать тут 
```suggestion
  Equotion curr {};
```
будет тоже самое